### PR TITLE
WebInspector infrastructure

### PR DESCRIPTION
**Pull request by PanayotCankov**
This will enable http requests to be profiled in the WebInspector for iOS
Rebased @KristinaKoeva /NetworkSpike, moved away some network domain details from the http modules, and added an intermediate module so that the debugger won't require http directly.

### CODE REVIEW

**Comment by PanayotCankov** [comment]
run-ci

**Comment by dtopuzov** [comment]
run ci

**Comment by dtopuzov** [comment]
run ci

**Comment by dtopuzov** [comment]
run ci

File: debugger/webinspector.ios.ts [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+
+    set resourceType(value: string) {
+        if (this._resourceType !== value) {
+                this._resourceType = value;
+        }
+    }
+    
+    public responseReceived(response: definitionType.NetworkDomain.Response): void {
+        if (this._networkDomainDebugger.enabled) {
+            this._networkDomainDebugger.events.responseReceived(this.requestID, "NativeScriptMainFrameIdentifier", "Loader Identifier", __inspectorTimestamp(), <any>this.resourceType, response);
```
Comment (hshristov) [inline debugger/webinspector.ios.ts]: `NativeScriptMainFrameIdentifier` & `Loader Identifier` better be const.

File: debugger/webinspector.ios.ts [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+    }
+    
+    public requestWillBeSent(request: definitionType.NetworkDomain.Request): void {
+        if (this._networkDomainDebugger.enabled) {
+            this._networkDomainDebugger.events.requestWillBeSent(this.requestID, "NativeScriptMainFrameIdentifier", "Loader Identifier", request.url, request, __inspectorTimestamp(), { type: 'Script' });
+        }
+    }
+}
+
+@definition.DomainDispatcher("Network")
```
Comment (hshristov) [inline debugger/webinspector.ios.ts]: `definition` is used in other modules to refere to `.d.ts`. Consider renaming it.

File: http/http-request.d.ts [hunk 3]
```diff
@@ -3,5 +3,6 @@
 declare module "http/http-request" {
     import http = require("http");
 
+    export var domainDebugger: any;
```
Comment (hshristov) [inline http/http-request.d.ts]: remove this

**Comment by hshristov** [comment]
:+1: after comments

**Comment by PanayotCankov** [comment]
run ci

**Comment by PanayotCankov** [comment]
run ci

**Pull request merged**